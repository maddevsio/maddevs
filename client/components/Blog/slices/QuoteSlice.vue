<template>
  <div class='post-part single'>
    <blockquote class="block-quotation" v-html="$prismic.asHtml(slice.primary.quote)" v-if="slice.slice_type === 'quote'"/>
    <text-quote v-if="slice.slice_type === 'quote_large'">{{ $prismic.asText(slice.primary.quote) }}</text-quote>
  </div>
</template>

<script>
import TextQuote from '../../Cases/TextQuote';
export default {
  components: {TextQuote},
  props: ['slice'],
  name: 'quote-slice'
};
</script>

<style lang="scss" scoped>
  @import '../../../assets/styles/_vars';

  .block-quotation {
    display: inline-block;
    font-style: italic;
    font-size: 17px;
    line-height: 28px;
    margin: 0 0 48px;
    font-family: 'Inter-Regular', sans-serif;
    color: $text-color--black-cases;
    border-left: 2px solid $border-color--red;
    padding-left: 35px;

    /deep/ p {
      margin-bottom: 28px;

      &:last-child {
        margin-bottom: 0;
      }
    }
  }

  .block-citation {
    display: inline-block;
    font-style: italic;
    border-left: solid #B4B4B4 4px;
  }
</style>
