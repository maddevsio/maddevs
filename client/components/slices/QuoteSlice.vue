<template>
  <div class="post-part single">
    <blockquote
      v-if="slice.primary.quotation_design === 'border'"
      class="block-quotation"
      v-html="$prismic.asHtml(slice.primary.quote)"
    />
    <TextQuote
      v-if="slice.primary.quotation_design === 'brackets'"
      class="post-quote"
    >
      {{ $prismic.asText(slice.primary.quote) }}
    </TextQuote>
    <div
      v-if="slice.primary.portrait_author.url || authorName"
      :class="`p-${slice.primary.quotation_design}`"
      class="author"
    >
      <div
        v-if="slice.primary.portrait_author.url"
        class="author__image"
      >
        <PrismicImage :field="slice.primary.portrait_author.mobile" />
      </div>
      <div
        v-if="authorName"
        class="author__name"
      >
        {{ authorName }}
      </div>
    </div>
  </div>
</template>

<script>
import TextQuote from '@/components/Cases/shared/TextQuote'

export default {
  name: 'QuoteSlice',
  components: {
    TextQuote,
  },

  props: {
    slice: {
      type: Object,
      required: true,
    },
  },

  computed: {
    authorName() {
      return this.$prismic.asText(this.slice.primary.name_of_the_author)
    },
  },
}
</script>

<style lang="scss" scoped>
.block-quotation {
  @include font('Inter', 17px, 400);
  display: inline-block;
  font-style: italic;
  line-height: 28px;
  margin: 24px 0;
  color: $text-color--black-oil;
  border-left: 2px solid $border-color--red;
  padding-left: 35px;

  /deep/ p {
    margin-bottom: 28px;
    &:last-child {
      margin-bottom: 0;
    }
  }
}

.block-citation {
  display: inline-block;
  font-style: italic;
  border-left: solid #b4b4b4 4px;
}

.author {
  display: flex;
  align-items: center;
  margin-bottom: 48px;

  &.p {
    &-border {
      padding-left: 37px;
    }
    &-brackets {
      padding-left: 62px;
    }
  }

  &__image {
    margin-right: 8px;
    img {
      width: 40px;
      min-width: 40px;
      height: 40px;
      border-radius: 50%;
      vertical-align: middle;
    }
  }

  &__name {
    @include font('Inter', 16px, 400);
    color: $text-color--grey-opacity-40-percent;
    font-style: italic;
    line-height: 166%;
    letter-spacing: -0.035em;
  }
}

/deep/ .post-quote {
  margin-left: 0;
  margin-right: 0;
}
</style>
