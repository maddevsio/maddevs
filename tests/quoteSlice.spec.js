import {mount} from '@vue/test-utils';
import QuoteSlice from '../client/components/Blog/slices/QuoteSlice';

describe('QuoteSlice component', () => {
  let wrapper;

  const slice = {
    primary: {
      quotation_design: 'border',
      quote: 'lorem ipsum dolor sit ame'
    }
  };

  beforeEach(() => {
    wrapper = mount(QuoteSlice, {
      propsData: {slice},
      mocks: {
        $prismic: {
          asText: text => text,
          asHtml: html => `<p>${html}</p>`
        }
      }
    });
  });

  test('is a Vue component', () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('renders text', () => {
    expect(wrapper.text()).toMatch(slice.primary.quote);
  });
});
