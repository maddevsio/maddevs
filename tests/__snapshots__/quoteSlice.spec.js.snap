// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteSlice component renders correctly 1`] = `
<div
  class="post-part single"
>
  <blockquote
    class="block-quotation"
  >
    <p>
      lorem ipsum dolor sit ame
    </p>
  </blockquote>
   
  <!---->
</div>
`;
